src/hotspot/share/opto/optoreg.hpp

```diff
@@ -184,27 +185,56 @@ class OptoReg {
 class OptoRegPair {
   friend class VMStructs;
 private:
-  short _second;
-  short _first;
+  typedef short Name;
+  Name _second;
+  Name _first;
+
 public:
-  void set_bad (                   ) { _second = OptoReg::Bad; _first = OptoReg::Bad; }
-  void set1    ( OptoReg::Name n  ) { _second = OptoReg::Bad; _first = n; }
-  void set2    ( OptoReg::Name n  ) { _second = n + 1;       _first = n; }
-  void set_pair( OptoReg::Name second, OptoReg::Name first    ) { _second= second;    _first= first; }
-  void set_ptr ( OptoReg::Name ptr ) {
+  static constexpr bool can_fit(OptoReg::Name n) {
+    return n <= std::numeric_limits<Name>::max();
+  }
```

Comment:
At first I thought this will always return true, but now I see it is checking OptoReg::Name against OptoRegPair::Name.

Reply:
I'll make it explicit and write `return n <= std::numeric_limits<OptoRegPair::Name>::max();` to avoid confusion.